### PR TITLE
feat(kubernetes): support apiVersion/Kind format in QueryResources

### LINKED ISSUE #1757: Support `apiVersion/Kind` format in QueryResources
`QueryResources` uses `GetClientByKind` which can't distinguish between resources with the same Kind but different API groups (e.g., `v1/Service` vs `serving.knative.dev/v1/Service`).

### Proposal

Support `apiVersion/Kind` format in the `Types` field:

- `v1/Pod`
- `apps/v1/Deployment`
- `serving.knative.dev/v1/Service`

When `/` is present, parse out apiVersion and Kind, then use `GetClientByGroupVersionKind` instead of `GetClientByKind`.

Existing usage without `/` remains unchanged for backward compatibility.

### PR DESCRIPTION
**Pull request by adityathebe**
Closes #1757

## Summary by CodeRabbit

* **New Features**
  * Parse Kubernetes apiVersion/kind strings in multiple formats (core, group, domain-based CRDs) and use them when selecting resources.

* **Improvements**
  * More robust client selection and REST-mapping handling for modern and legacy API forms, improved cache integration and logging, and clearer error messages referencing the apiVersion/kind.

* **Tests**
  * Added comprehensive parsing tests covering valid inputs, edge cases, and invalid formats.

### CODE REVIEW

File: kubernetes/dynamic_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+		{
+			name:            "custom resource definition",
+			input:           "cert-manager.io/v1/Certificate",
+			expectedGroup:   "cert-manager.io",
+			expectedVersion: "v1",
+			expectedKind:    "Certificate",
+			expectError:     false,
+		},
+		{
+			name:        "invalid format - no slash",
```
[the remark this answers is not in the document]
Comment (adityathebe) [reply kubernetes/dynamic_test.go]: The existing logic is to support types like `Pod`. This change is backward compatible with that.



We only use this new format/parser if the type has `/` in it.
https://github.com/flanksource/duty/pull/1758/changes/BASE..c68b4013b52a48de12e84aefb94e55d87d5fef62#diff-71455d5eeb1300adc7e8c16a7672b2a3f38035d3022d4a55471addcc097cdca3R557
